packages/fxa-content-server/tests/functional/oauth_webchannel.js

```diff
@@ -0,0 +1,126 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+const { registerSuite } = intern.getInterface('object');
+const TestHelpers = require('../lib/helpers');
+const FunctionalHelpers = require('./lib/helpers');
+const selectors = require('./lib/selectors');
+
+const SERVICE_NAME = '123done';
+const PASSWORD = 'passwordzxcv';
+
+let email;
+
+const {
+  click,
+  closeCurrentWindow,
+  createUser,
+  fillOutSignIn,
+  fillOutSignUp,
+  openFxaFromRp,
+  openVerificationLinkInNewTab,
+  switchToWindow,
+  testElementExists,
+  testElementTextInclude,
+  testIsBrowserNotified,
+  testUrlInclude,
+} = FunctionalHelpers;
+
+registerSuite('oauth webchannel', {
+  beforeEach: function() {
+    email = TestHelpers.createEmail();
+
+    return this.remote.then(
+      FunctionalHelpers.clearBrowserState({
+        '123done': true,
+        contentServer: true,
+        force: true,
+      })
+    );
+  },
+  tests: {
+    signup: function() {
```

Comment:
Does Fenix make use of the email first flow? If so, should probably test that too.

Reply:
https://github.com/mozilla/fxa/issues/2415